### DISCUSSION TITLE
Possible to not read last names?

### DISCUSSION THREAD

**Discussion by FragSauce** [karashiiro/TextToTalk#85] [Q&A] [answered]
Hey there i really like this plugin, and i like to know who is talking since i read slow, but having to hear everyones last name takes quite long, would it be possible to get another option under the "Enable "X says: when people speak" to not include the last name and just the first name?

**Comment by ryankhart** [comment, marked as answer]
Hey @FragSauce. If you want to get notifications about your requested feature, you can subscribe to notifications on this [issue](https://github.com/karashiiro/TextToTalk/issues/87). I'm currently working on it right now.

**Comment by gucciBane** [comment]
Is it possible to piggyback this with excluding last name, or more importantly, world name, from players' names? Hearing 'Jesse James Coeurl' over and over in Cross World Parties can get pretty redundant.

Thanks for the great plugin!

**Reply by ryankhart** [reply]
I created a new issue [here](https://github.com/karashiiro/TextToTalk/issues/108) for your idea. Good idea! I don't know how I never noticed that before. It might just be because I'm only hearing it when I'm way too distracted and feeling obligated to go fast (in dungeons). I'll need to wait until I've run some duty finder instances and have a log to go through and figure out how to parse it properly.

Follow this issue for future updates and discussion.
https://github.com/karashiiro/TextToTalk/issues/108

**Reply by gucciBane** [reply]
Awesome! Yeah, at first I didn't realize why everyone had such bizarre long names. It would definitely shorten consecutive calls if it wasn't announcing everyone's address haha.

Thanks!

**Reply by ryankhart** [reply]
Hmm, I wonder if we could detect when the player is in a duty instance and optionally only shorten names in duties. After implementing the only first or last names feature, I tried it out for while, but I eventually turned it back off for me because I missed hearing the full names of characters in my FC, but I'd still want a first names only feature while in duties. And I could nest the option(s) for that in the already implemented collapsible setting for speaking shorter names.
https://github.com/karashiiro/TextToTalk/issues/109

**Comment by gucciBane** [comment]
That would be fine, I think. That would really only exclude hunt trains.

**Reply by ryankhart** [reply]
Oh, good point. Hunts are one activity that have yet to try out yet. Maybe limiting it to duties is too specific. Maybe it should be an option to shorten names spoken for party and alliance chat. I think that would cover all of my bases for what I'd want from the new feature.

#109 for reference.
